### ISSUE TITLE
Training Loss: Loss seems to have plateaued over time and seems to have random spike in losses.

### ISSUE THREAD

**Issue by VarnithChordia** [TIGER-AI-Lab/VLM2Vec#128]
I am trying to reproduce your results for V2 (including images and videos as seen [here](https://github.com/TIGER-AI-Lab/VLM2Vec/blob/main/experiments/public/train/train_alltasks.yaml)). I am training the QWEN2VL-3b with similar training hyper-parameters as provided in the scripts. I am observing that loss has plateaued and also occasional spikes in the loss. 

WDYT?  Did you observe something similar while training as well?

The below is  plot of training until 1300 steps

<img width="1174" height="631" alt="Image" src="https://github.com/user-attachments/assets/96bd1f11-6845-421d-8a2c-3a2d4c7f600f" />

**Comment by memray** [comment]
Hi @VarnithChordia ,

The curve in the figure looks good to me. The spikes in the training loss is kinda common due to the uneven difficulty across sampled batches (e.g. image tasks are supposedly easier than retrieval). As long as the moving average goes down, it should be fine. Hope this helps.

Best,
Rui

**Comment by XMHZZ2018** [comment]
I will close this issue for now. If any further issues arise, feel free to reopen it. Thanks!
